packages/zowe-explorer/__tests__/__unit__/trees/dataset/DatasetTree.unit.test.ts

```diff
@@ -3010,9 +3010,19 @@ describe("Dataset Tree Unit Tests - Sorting and Filtering operations", () => {
     });
 
     describe("addDsTemplate", () => {
-        it("adds a new DS template to the persistent object", () => {
-            tree.addDsTemplate({ test: "test" } as any);
-            expect(tree["mHistory"]["mDsTemplates"].length).toEqual(1);
+        it("adds a new DS template to the persistent object", async () => {
+            const mockTemplates = [{ test1: {} }, { test2: {} }, { test3: {} }];
+            const newTemplate = { test: {} };
+            mockTemplates.unshift(newTemplate as any);
+            jest.spyOn(vscode.workspace, "getConfiguration").mockReturnValue({
+                inspect: jest.fn().mockReturnValue({ globalValue: mockTemplates }),
+            } as any);
+            Object.defineProperty(vscode.workspace, "workspaceFolders", { value: [], configurable: true });
+            const infoLoggerSpy = jest.spyOn(ZoweLogger, "info");
+            Object.defineProperty(SettingsConfig, "setDirectValue", { value: jest.fn(), configurable: true });
+
+            await tree.addDsTemplate(newTemplate as any);
+            expect(infoLoggerSpy).toHaveBeenCalledWith("Adding new data set template {0}.");
```

Comment:
In addition to testing that an info message is shown, should we test that a function to update the setting was called? Either `DataSetTemplates.updateDsTemplateSetting` or `SettingsConfig.setDirectValue`.